core/src/main/java/org/elasticsearch/action/percolate/TransportPercolateAction.java

```diff
@@ -196,9 +196,8 @@ public static SearchRequest createSearchRequest(PercolateRequest percolateReques
 
         PercolatorQueryBuilder percolatorQueryBuilder = new PercolatorQueryBuilder(percolateRequest.documentType(), documentSource);
         if (querySource != null) {
-            QueryParseContext queryParseContext = new QueryParseContext(queryRegistry);
-            queryParseContext.reset(XContentHelper.createParser(querySource));
-            queryParseContext.parseFieldMatcher(parseFieldMatcher);
+            QueryParseContext queryParseContext = new QueryParseContext(queryRegistry, XContentHelper.createParser(querySource),
```

Comment:
Sure, I see parse creation a lot as try-with-resouces without a finally block (unless we need to reset something which here I think we don't), so I will do that.